frontend/projects/ui/src/app/pages/server-routes/marketplaces/marketplaces.page.ts

```diff
@@ -188,11 +189,11 @@ export class MarketplacesPage {
       .subscribe()
   }
 
-  private async delete(id: string): Promise<void> {
+  private async delete(id?: string): Promise<void> {
     if (!id) return
-    const marketplace: UIMarketplaceData = JSON.parse(
-      JSON.stringify(this.patch.getData().ui.marketplace),
-    )
+
+    const data = await getMarketplace(this.patch)
+    const marketplace: UIMarketplaceData = JSON.parse(JSON.stringify(data))
```

Comment:
since we are using this JSON.parse/stringify a few times, can we make it a function with descriptive name? I'm not sure _why_ we need to do this, do you know @MattDHill ?